File: Algorithm/3_LongestSubstringWIthoutRepeatingCharacters/main.py
def lengthOfLongestSubstring(self, s: str) -> int:
    temp = ""
    output = 0
    for i in range(len(s)):
        index = temp.find(s[i])
        if(index >= 0):
            temp = temp[index+1:]
            print(temp)
        temp += s[i]
        output = max(output, len(temp))
    return output


s = "1232453"
print(lengthOfLongestSubstring(0, s))

File: Algorithm/3_LongestSubstringWIthoutRepeatingCharacters/main.cpp
#include <bits/stdc++.h>
using namespace std;
int lengthOfLongestSubstring(string s)
{
    string temp = "";
    int max = 0;
    for (int i = 0; i < s.size(); i++)
    {
        if (temp.find(s[i]) < temp.size())
        {
            size_t position = temp.find(s[i]);
            temp = temp.substr(position + 1);
        }
        temp += s[i];
        int size = temp.size();
        if (max < size)
        {
            max = size;
        }
    }
    return max;
}
int main()
{
    string a = "abcad";
    cout << lengthOfLongestSubstring(a) << endl;

    return 0;
}